src/main/integrations/slack/index.ts

```diff
@@ -0,0 +1,81 @@
+import playerStateStore, { PlayerState, VideoDetails } from "../../player-state-store";
+import IIntegration from "../integration";
+import { StoreSchema } from "~shared/store/schema";
+import Conf from "conf";
+
+export default class SlackStatus implements IIntegration {
+  private store: Conf<StoreSchema>;
+  private enabled = false;
+  private stateCallback: (event: PlayerState) => void = null;
+  private currentVideoId: string = "";
+  private statusTimeoutId: NodeJS.Timeout;
+  private slackUserToken: string = "";
+
+  private async playerStateChanged(state: PlayerState) {
+    const { secondsTillClearStatus } = await this.store.get("slack");
+    if (state.trackState === 1 && state.videoDetails) {
+      if (!state.adPlaying) {
+        if (this.statusTimeoutId) clearTimeout(this.statusTimeoutId);
+        if (this.currentVideoId === state.videoDetails.id) {
+          this.statusTimeoutId = setTimeout(() => {
+            this.currentVideoId = "";
+            this.updateSlackStatus(true, state.videoDetails);
+          }, secondsTillClearStatus * 1000);
+        } else {
+          this.currentVideoId = state.videoDetails.id;
+          this.updateSlackStatus(false, state.videoDetails);
+        }
+      }
+    }
+  }
+  private updateSlackStatus(clear: boolean, data: VideoDetails) {
+    const message = `Listening To: ${data.title} | ${data.author}`;
+    let postData: { status_text: string; status_emoji: string; status_expiration: number } = {
+      status_text: "",
+      status_emoji: "",
+      status_expiration: 0
+    };
+    if (!clear) {
+      const currentUnixTimestamp = Math.floor(Date.now() / 1000);
```

Comment:
`currentUnixTimestamp` and `message` feel like they're not really needed as variables if they are only storing stuff for a single usage within the same function?

I would personally just put them directly inside of the `postData`.

Reply:
the timestamp can be moved from 2 variable to 1 and message can be removed and used without a variable. this has been reflected in my latest push

Reply:
The way slack works though is that you must send empty data to clear the status. any data inside will create a status. I have instead created 2 separate post data variables, one with blank data that clears, one with real data that creates a status